src/org/zaproxy/zap/extension/pscanrulesAlpha/XDebugTokenScanner.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.Vector;
+
+import net.htmlparser.jericho.Source;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+/**
+ * X-Debug-Token passive scan rule
+ * https://github.com/zaproxy/zaproxy/issues/2452
+ * 
+ * @author kingthorin+owaspzap@gmail.com
+ */
+public class XDebugTokenScanner extends PluginPassiveScanner {
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.xdebugtoken.";
+	private static final int PLUGIN_ID = 10056;
+
+	private PassiveScanThread parent = null;
+	private static final Logger LOGGER = Logger.getLogger(XDebugTokenScanner.class);
+
+	private static final String X_DEBUG_TOKEN_HEADER = "X-Debug-Token";
+	private static final String X_DEBUG_TOKEN_LINK_HEADER = "X-Debug-Token-Link";
+	private static final String PROFILER_PORTION = "_profiler"; // http://example.com/_profiler/c6bd78
+
+	@Override
+	public void setParent(PassiveScanThread parent) {
+		this.parent = parent;
+	}
+
+	@Override
+	public void scanHttpRequestSend(HttpMessage msg, int id) {
+		// Only checking the response for this plugin
+	}
+
+	@Override
+	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		long start = System.currentTimeMillis();
+
+		//Check "Link" variant first as it's of greater concern/convenience.
+		Vector<String> xdtlHeaders = msg.getResponseHeader().getHeaders(X_DEBUG_TOKEN_LINK_HEADER);
+		if (xdtlHeaders != null) { // Header Found
+			for (String xdtl : xdtlHeaders) {
+				parent.raiseAlert(id, getAlert(msg, X_DEBUG_TOKEN_LINK_HEADER));

```

Comment:
The value of the header is not being used (the link is being constructed with the header name instead of the value).